include/alpaka/kernel/TaskKernelGpuUniformCudaHipRt.hpp

```diff
@@ -272,118 +281,17 @@ namespace alpaka
                     },
                     task.m_args);
 
-                if constexpr(ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL)
+                if constexpr(TBlocking || ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL)
```

Comment:
[nit] may be adding a aline explaining the TBlocking check could be good.

Reply:
The comment two lines below is pretty explanatory: "Wait for the kernel execution to finish (etc.)"

Like the old code did not comment
```c++
// if debugging is enabled, wait
if constexpr(ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL)
  {
  ...
```
I think that adding a comment like
```c++
// if the queue is blocking or debugging is enabled, wait
if constexpr(TBlocking || ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL)
  {
  ...
```
is a bit redundant.

Reply:
By the way, I'm considering a further change:
  - if the queue is blocking, synchronise and check the result of the synchronisation
  - if the queue is non-blocking, do not synchronise, and check only the result of the kernel submission

both when debugging is enabled or disabled.

However I would prefer to keep this PR one that does not introduce any changes in behaviour, and suggest this change in a separate PR.